### PR TITLE
Gift card updates

### PR DESCRIPTION
**Pull request by jowlo**
Fixes some smaller visual glitches during gift card signup and gift card redemption.

Add link to gift card terms and faq below list of gift cards (also shows payment methods)

### CODE REVIEW

**Comment by jowlo** [comment]
fixes #2497

File: src/subscription/giftcards/RedeemGiftCardWizard.js [hunk 1]
```diff
@@ -394,6 +392,12 @@ export function loadRedeemGiftCardWizard(giftCardInfo: GiftCardRedeemGetReturn,
 				componentClass: RedeemGiftCardPage
 			}
 		]
-		return createWizardDialog(giftCardRedeemData, wizardPages, () => Promise.resolve(m.route.set("/login"))).dialog
+		return createWizardDialog(giftCardRedeemData, wizardPages,
+			() => {
+				const route = giftCardRedeemData.credentialsMethod === "signup"
+					? `/login?loginWith=${giftCardRedeemData.mailAddress()}`
```
Comment (charlag) [inline src/subscription/giftcards/RedeemGiftCardWizard.js:398]: I don't like string manipulation in url, here's how we do it usually:
https://github.com/tutao/tutanota/blob/7f8fe5569829be3bcce5c47d5febad9fa6f7a7f8/src/login/RecoverLoginDialog.js#L125
mithril will `encodeURIComponent()` as needed

File: src/misc/TranslationKey.js [hunk 2]
```diff
@@ -473,8 +473,8 @@ export type TranslationKeyType = "about_label"
 	| "giftCardRedeemed_msg"
```
Comment (charlag) [inline src/misc/TranslationKey.js:473]: You removed this translation in code but not in Phrase and it got pulled in again

File: src/subscription/giftcards/RedeemGiftCardWizard.js (hunk 1 above)
Comment (jowlo) [reply src/subscription/giftcards/RedeemGiftCardWizard.js:398]: Thanks, much nicer!

File: src/misc/TranslationKey.js (hunk 2 above)
Comment (jowlo) [reply src/misc/TranslationKey.js:473]: Done.

**Review by charlag — APPROVED** [review]

**Pull request merged**